hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java

```diff
@@ -156,11 +160,54 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
     /**
      * Validate the fields in the transaction body that involves checking with dynamic
      * properties or state. This check is done as part of the handle workflow.
-     * @return OK if the transaction body is valid, otherwise return the appropriate error code
+     * @param context handle context
+     * @param accountStore account store
+     * @param op crypto create transaction body
+     * @return modified payer account
      */
-    private ResponseCodeEnum validateSemantics() {
-        // FUTURE : Need to add validations that involve dynamic properties or state
-        return OK;
+    private Account validateSemantics(
+            @NonNull final HandleContext context,
+            @NonNull final WritableAccountStore accountStore,
+            @NonNull final CryptoCreateTransactionBody op) {
+        final var cryptoCreateWithAliasConfig =
+                context.configuration().getConfigData(CryptoCreateWithAliasConfig.class);
+        final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
+        final var entitiesConfig = context.configuration().getConfigData(EntitiesConfig.class);
+        final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+
+        // validate payer account exists and has enough balance
+        final var payer = accountStore.getForModify(
+                context.body().transactionIDOrElse(TransactionID.DEFAULT).accountIDOrElse(AccountID.DEFAULT));
+
+        validateTrue(payer != null, INVALID_PAYER_ACCOUNT_ID);
```

Comment:
@Neeharika-Sompalli do we need `getIfUsable` here instead of only checking for null?

Reply:
yes agreed @mhess-swl

Reply:
@iwsimon there's a method in `TokenHandlerHelper` called `getIfUsable(AccountID, ...)` that will check for null, paused, deleted, and expired accounts. 

E.g.
```java
final var payer = TokenHandlerHelper.getIfUsable(accountId, accountStore, context.expiryValidator(), INVALID_ACCOUNT_ID);
```

Reply:
Added getIfUsable() check, still need original validateFalse(payer == null, INVALID_PAYER_ACCOUNT_ID) because accountId can't be null in getIfUsable().